### PR TITLE
Update package-lock.json files in publish.sh

### PR DESCRIPTION
**Pull request by callumacrae**
By default lerna doesn't seem to handle this - this runs npm install in
the packages which updates the package-lock.json files, then adds them
to the previous commit and pushed to github.

Not doing this caused a problem for me where if you run publish.sh twice
without something else happening to update the lockfiles, lerna fails to
publish because the npm install in publish.sh updates the lockfiles and
lerna won't run on a non-empty working tree.

### CODE REVIEW

File: publish.sh [hunk 1]
```diff
@@ -13,3 +13,8 @@ git checkout -b $publish_branch
 git push origin $publish_branch
 
 npx lerna publish
+
+# Update package-lock.json files as well
+npx lerna exec npm install
```
Comment (meraymond2) [inline publish.sh]: This is already run in line 5, if the only new line is the commit line up there?

Comment (callumacrae) [reply publish.sh]: - The first `npm install` makes sure everything is up to date, but usually doesn't do anything—if it does, `lerna publish` will fail as the working tree won't be empty
- `lerna publish` updates the version numbers in the package.json files and commits, but doesn't run `npm install` so the lockfiles remain unchanged
- The second `npm install` updates the lockfiles

File: publish.sh [hunk 2]
```diff
@@ -13,3 +13,8 @@ git checkout -b $publish_branch
...
+git commit --all --amend --no-edit
```
Comment (brendonboshell) [inline publish.sh]: Does lerna not tie tags to a particular commit? Would this not destroy the tags?

**Comment by brendonboshell** [comment]
It looks like if we run `lerna version` explicitly, there are lifecycle scripts we can hook onto: https://github.com/lerna/lerna/tree/master/commands/version#lifecycle-scripts

Is it possible to use the `postversion` script to update package-lock.json

**Comment by brendonboshell** [comment]
> It looks like if we run `lerna version` explicitly, there are lifecycle scripts we can hook onto: https://github.com/lerna/lerna/tree/master/commands/version#lifecycle-scripts
> 
> Is it possible to use the `version` script to update package-lock.json

The `version` lifecycle script also runs from the root `package.json`.

**Comment by callumacrae** [comment]
That looks like a much nicer solution. Testing now.

File: package.json [hunk 3]
```diff
@@ -10,6 +10,7 @@
     "start": "start-storybook -p 6006",
     "--------------- BUILDING": "----------------------",
     "build-storybook": "build-storybook",
+    "version": "lerna exec \"npm install --ignore-scripts --package-lock-only\" && git add .",
```
[the remark this answers is not in the document]
Comment (callumacrae) [reply package.json]: I'm not sure. I've looked through the source and it seems to be adding specific files, not all changed files, but there is some lockfile related stuff in there too.

Perhaps I could remove it and merge, and then if it doesn't work when you deploy your change I could add it back in another PR? Can't really think of a better way to test this as I have nothing left to publish.

[the remark this answers is not in the document]
Comment (brendonboshell) [reply package.json]: Cool, I don't think we need to remove it. Approved

**Pull request merged**